### PR TITLE
feat(fclasses): Add configurable attribute for design keys

### PR DESCRIPTION
**Pull request by SW999**
Issue: https://github.com/johnsonandjohnson/Bodiless-JS/issues/765

## Changes

* Renamed env variable from **BODILESS_DEBUG** to **BODILESS_SHOWDESIGNKEYS**
* **extendDesignable** can get design keys attribute name from context
* Updated **addPropsIf**: **propsToAdd** may be an object or function which returns object
* Updated HOC **withShowDesignKeys**. At the moment the Page component has design keys from default context, but we can alter them using **withShowDesignKeys(true, 'test-attr')(Fragment)** as a wrapper for the Page component on a site level.

### CODE REVIEW

File: packages/fclasses/__tests__/Design.test.tsx [hunk 1]
```diff
@@ -169,19 +169,39 @@ describe('withShowDesignKeys', () => {
   it('Adds design keys when enabled', () => {
     const Test: ComponentType<any> = flow(
       designable(startComponents, 'Base'),
-      withShowDesignKeys(),
+      withShowDesignKeys({}),
```
Comment (beliayeu) [inline packages/fclasses/__tests__/Design.test.tsx:172]: consider allowing execution of this function without params (the function will take default one)

File: packages/fclasses/src/Context.tsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import React, { createContext, ComponentType, useContext } from 'react';
 
 type FClassesContextType = {
   showDesignKeys?: boolean,
+  dataLayerAttribute?: string;
```
Comment (beliayeu) [inline packages/fclasses/src/Context.tsx:5]: not sure we need to use dataLayer term here in @bodiless/fclasses
consider renaming this variable

File: packages/fclasses/src/Context.tsx [hunk 3]
```diff
@@ -11,14 +12,19 @@ const FClassesContext = createContext<FClassesContextType>({});
  * all children.
  *
  * @param showDesignKey true to enable (the default), false to disable.
+ * @param {string} [dataLayerAttribute].
```
Comment (beliayeu) [inline packages/fclasses/src/Context.tsx:15]: not sure this jsdoc is correct

File: packages/fclasses/src/Design.tsx [hunk 4]
```diff
@@ -296,7 +296,7 @@ export const extendDesignable = (transformDesign: TransformDesign = identity) =>
       const designKeys = typeof start !== 'function'
         ? Object.keys(start).reduce((keys, key) => ({
           ...keys,
-          [key]: addPropsIf(useShowDesignKeys)({ 'data-bl-design-key': `${namespace}:${key}` }),
+          [key]: addPropConditionally(useShowDesignKeys, useDataLayerAttribute)(`${namespace}:${key}`),
```
Comment (beliayeu) [inline packages/fclasses/src/Design.tsx:299]: discuss refactoring this

File: packages/fclasses/src/Context.tsx (hunk 2 above)
Comment (SW999) [reply packages/fclasses/src/Context.tsx:5]: What about 'designKeysAttributeName'?

File: packages/fclasses/src/Context.tsx (hunk 3 above)
Comment (SW999) [reply packages/fclasses/src/Context.tsx:15]: I suppose this parameter will be optional and returns a string value. Please check https://jsdoc.app/tags-param.html#optional-parameters-and-default-values

File: packages/fclasses/__tests__/Design.test.tsx (hunk 1 above)
Comment (SW999) [reply packages/fclasses/__tests__/Design.test.tsx:172]: Will change

File: packages/fclasses/src/Design.tsx (hunk 4 above)
Comment (SW999) [reply packages/fclasses/src/Design.tsx]: Updated. Please check

**Review by beliayeu — APPROVED** [review]

**Review by beliayeu — APPROVED** [review]

**Review by beliayeu — APPROVED** [review]

**Review by awm086 — APPROVED** [review]

**Review by hvanyo — COMMENTED** [review]
Before merging please update title to include scope (fclasses). -- https://johnsonandjohnson.github.io/Bodiless-JS/#/Development/Contributing?id=merging-pull-request

**Pull request merged**